lib/cisco_node_utils/cmd_ref/bridge_domain.yaml

```diff
@@ -3,11 +3,54 @@
 ---
 _exclude: [N3k, N5k, N6k, N8k, N9k, ios_xr]
 
+# For the below name, shutdown commands we are going to use show running-config
+# bridge-domain cli which displays as below
+#
+# system bridge-domain 100-110
+#
+# bridge-domain 100-110
+#
+# bridge-domain 100
+#   shutdown
+#   fabric-control
+#   name PepsiCo
+# bridge-domain 101
+# bridge-domain 102
+# ...
+# bridge-domain 108
+#   shutdown
+#   name Amazon
+# bridge-domain 109
+#   shutdown
+# bridge-domain 110
+#   name Microsoft
+
+_template:
+  get_command: "show running-config bridge-domain"
+
 all_bds:
   multiple: true
-  get_command: "show running-config bridge-domain"
   get_value: '/^bridge-domain (\S+)/'
 
+# Here for name match if condition is used because the underlying property is
+# system bridge-domain 100
+#
+# bridge-domain 100
+#
+# bridge-domain 100
+#   shutdown
+#   fabric-control
+#   name PepsiCo
+# so we cannot use get_context: '/^bridge-domain 100/' because it will search
+# for the first context and there is nothing under that. The properties are
+# present in the second context. Also name could appear in any order and hence
+# the if conditions.
+bd_name:
+  get_value: '/^bridge-domain[\s\S]*bridge-domain <bd>\s*(?:shutdown\s+fabric-control\s+name (\S+)|fabric-control\s+name (\S+)|shutdown\s+name (\S+)|name (\S+))/'
```

Comment:
Ugh. This regex could be a lot simpler if our command ref infrastructure could handle multiline matching. But we'll save that for another day... :-(

In this case you can still simplify a bit by doing this (still ugly but a little better) pattern:

```
  get_value: '/bridge-domain <bd>\s*.*\s*.*\s*name (\S+)/
```